internal/pkg/cli/app_show.go

```diff
@@ -129,24 +132,45 @@ func (o *showAppOpts) description() (*describe.App, error) {
 	if err != nil {
 		return nil, fmt.Errorf("list jobs in application %s: %w", o.name, err)
 	}
-	//1st approach takes 2 seconds to display the table
-	for _, env := range envs {
-		deployedJobs, err := o.deployStore.ListDeployedJobs(o.name, env.Name)
-		if err != nil {
-			return nil, fmt.Errorf("list deployed jobs %s: %w", o.name, err)
-		}
-		for _, job := range deployedJobs {
-			workloadEnvs[job] = append(workloadEnvs[job], env.Name)
-		}
-		deployedSvcs, err := o.deployStore.ListDeployedServices(o.name, env.Name)
-		if err != nil {
-			return nil, fmt.Errorf("list deployed services %s: %w", o.name, err)
-		}
-		for _, svc := range deployedSvcs {
-			workloadEnvs[svc] = append(workloadEnvs[svc], env.Name)
-		}
-	}
+	var mux sync.Mutex
+	g, ctx := errgroup.WithContext(context.Background())
```

Comment:
Should we use a context with a deadline to ensure we return a response back to users within 30s for example? https://pkg.go.dev/context#example-WithDeadline